pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -96,6 +96,8 @@ public RealtimeSegmentDataManager(final RealtimeSegmentZKMetadata segmentMetadat
       RealtimeTableDataManager realtimeResourceManager, final String resourceDataDir, final ReadMode mode,
       final Schema schema, final ServerMetrics serverMetrics) throws Exception {
     super();
+    final String segmentVersionStr = tableConfig.getIndexingConfig().getSegmentFormatVersion();
```

Comment:
I meant that as when we load the indexing config from ZK/Helix, what happens if that property is not there or has an invalid value stored? Is it a mandatory property and Jackson gives an error, is it null or does it get typecasted into the right type?